### PR TITLE
Adds totp rate limit

### PR DESCRIPTION
**Pull request by ValJed**
[IN-22](https://apostrophecms.atlassian.net/browse/IN-22)

## Purpose
Extracting the throttle system from the classic login logic to reuse it for the login totp logic.
Also avoid that servers breaks if we call the `/login-totp` route without any user in **req**.

We use the throttle system already used for classic login.
I extracted this logic into three methods:
* checkThrottleAttempts: Verify before trying to log the user that he didn't already exceed the throttle limit.
* addThrottleAttempt: During login fails, it adds an attempt to the user and checks if he should be locked or not.
* cleanUpThrottleAttempts: Clean old attempts if we are in a login fail, if the login succeed we remove all attempts (to avoid any conflicts between classic login attempts and TOTP attempts).

@boutell for the redirection issue, for now I just added the `/modules/apostrophe-areas/editor` path to the safeList. It fixed the problem even if definitely not ideal.
Not sure about your `404 things matching that pattern`, if you want me to implement it this way let me know.

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]
Please see the existing `throttle` option to the login module. I would suggest that this should apply equally to regular login attempts and TOTP attempts, rather than introducing a separate mechanism and a separate syntax for configuration for TOTP throttling only.

File: lib/modules/apostrophe-login/index.js [hunk 1]
```diff
@@ -428,40 +404,102 @@ module.exports = {
...
-        if (userSafe.attempts && userSafe.attempts.filter(attempt => (attempt >= perSince)).length >= allowedAttempts) {
-          await safe.update({
-            _id: user._id
-          }, {
-            $set: {
-              lockout: new Date()
-            }
-          });
-          return true;
+        const unallowedAttemps = userSafe.attempts && userSafe.attempts
```
Comment (boutell) [inline lib/modules/apostrophe-login/index.js:409]: Typo: should be `unallowedAttempts` and `disallowedAttempts` would be better grammar. Actally let's just change it to `disallowed`, no need for the Attempts part of this variable name.

**Review by boutell — CHANGES_REQUESTED** [review]
One minor change.

Has this been thoroughly tested for both regular login throttling and TOTP throttling?

File: lib/modules/apostrophe-login/index.js (hunk 1 above)
Comment (ValJed) [reply lib/modules/apostrophe-login/index.js]: Yes it's clearer!

**Comment by ValJed** [comment]
@boutell Yes I did for both, and I made sure that the attempts are independent between `login` and `login-totp` by cleaning them when login.
You can test it easily in a boilerplate if you want to.

**Review by boutell — APPROVED** [review]

**Comment by boutell** [comment]
Tested successfully. Nice work!

**Comment by ValJed** [comment]
I just removed the `express-rate-limit` package and merged 2.0 updates.

**Review by boutell — APPROVED** [review]

**Pull request merged**